src/gen10_vdenc_vp9.h

```diff
@@ -0,0 +1,1644 @@
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Authors:
+ *    Xiang Haihao <haihao.xiang@intel.com>
+ *
+ */
+
+#ifndef GEN10_VDENC_VP9_H
+#define GEN10_VDENC_VP9_H
+
+#include <drm.h>
+#include <i915_drm.h>
+#include <intel_bufmgr.h>
+
+#include "i965_gpe_utils.h"
+#include "i965_encoder.h"
+#include "vp9_probs.h"
+
+struct encode_state;
+
+#define VP9_FRAME_I                             0
+#define VP9_FRAME_P                             1
+
+#define VP9_REF_NONE                            0x00
+#define VP9_REF_LAST                            0x01
+#define VP9_REF_GOLDEN                          0x02
+#define VP9_REF_ALT                             0x04
+
+#define VP9_SUPER_BLOCK_WIDTH                   64
+#define VP9_SUPER_BLOCK_HEIGHT                  64
+#define VP9_MI_UNIT_SIZE                        8
+
+#define VP9_MAX_SEGMENTS                        8
+
+#define VDENC_VP9_HUC_BRC_INIT_RESET            11
+#define VDENC_VP9_HUC_BRC_UPDATE                12
+#define VDENC_VP9_HUC_PROB                      13
+#define VDENC_VP9_HUC_INITIALIZER               14
+
+#define VDENC_VP9_HUC_DMEM_DATA_OFFSET          0x2000
+#define VDENC_VP9_HUC_SUPERFRAME_PASS           2
+
+#define VDENC_VP9_BRC_STATS_BUF_SIZE                    (48 * sizeof(unsigned int))
+#define VDENC_VP9_BRC_PAK_STATS_BUF_SIZE                (64 * sizeof(unsigned int))
+#define VDENC_VP9_BRC_HISTORY_BUFFER_SIZE               1152
+
+#define VDENC_VP9_BRC_MAX_NUM_OF_PASSES                 3
+#define VDENC_VP9_BRC_CONSTANT_DATA_SIZE                1664
+
+#define VDENC_VP9_BRC_BITSTREAM_SIZE_BUFFER_SIZE        64
+#define VDENC_VP9_BRC_BITSTREAM_BYTE_COUNT_OFFSET       0
+
+#define VDENC_VP9_BRC_HUC_DATA_BUFFER_SIZE              64
+
+#define VDENC_VP9_HUC_DATA_EXTENSION_SIZE               32
+
+#define VDENC_HCP_VP9_PIC_STATE_SIZE                    168
+#define VDENC_HCP_VP9_SEGMENT_STATE_SIZE                32
+#define VDENC_VDENC_CMD0_STATE_SIZE                     120
+#define VDENC_VDENC_CMD1_STATE_SIZE                     148
+#define VDENC_BATCHBUFFER_END_SIZE                      4       /* END */
+
+#define VDENC_HCP_VP9_PIC_STATE_SIZE_IN_DWS             (VDENC_HCP_VP9_PIC_STATE_SIZE >> 2)
+#define VDENC_HCP_VP9_SEGMENT_STATE_SIZE_IN_DWS         (VDENC_HCP_VP9_SEGMENT_STATE_SIZE >> 2)
+#define VDENC_VDENC_CMD0_STATE_SIZE_IN_DWS              (VDENC_VDENC_CMD0_STATE_SIZE >> 2)
+#define VDENC_VDENC_CMD1_STATE_SIZE_IN_DWS              (VDENC_VDENC_CMD1_STATE_SIZE >> 2)
+
+enum VDENC_VP9_LUT_MODE {
+    VDENC_VP9_LUT_MODE_INTRA_SADMPM             = 0x00,
+    VDENC_VP9_LUT_MODE_INTRA_32x32              = 0x01,
+    VDENC_VP9_LUT_MODE_INTRA_16x16              = 0x02,
+    VDENC_VP9_LUT_MODE_INTRA_8x8                = 0x03,
+
+    VDENC_VP9_LUT_MODE_INTER_32x16              = 0x04,
```

Comment:
these three emum value use the same 0x04. Is it reasonable?

Reply:
This enum isn't used any more, I will remove it from this commit.